src/model/table.py

```diff
@@ -20,8 +40,49 @@ def set(self):
     def delete(self):
         pass
 
-    def get(self):
-        pass
+    def get(self, query):
+        best_search = self.__get_efficient_primary_keys__(query)
+        found_objects = self.get_rows(best_search)
+        return self.__compare_found__(found_objects, query)
 
-    def get_by_primary_key(self):
-        pass
+    def __get_efficient_primary_keys__(self, query):
+        index_names = self.__table_metadata__.get_indices_names()
+        primary_keys = []
+        if query is not None or query != "" or not str(query).isspace:
```

Comment:
Same above, use is space

Reply:
You get efficient index not efficient primary keys

Reply:
This method is too complicated. I won’t review it. Remove it

Reply:
Don’t merge the logic of get efficient index by get the corresponing keys
Let this method focus only for returning the best index